File: perl/streaming_pipe_in_and_out.pl
use strict;
use warnings;

open my $output, '>', 'streaming-out.pipe';
open my $input, '<', 'streaming-in.pipe';

$output->autoflush(1); # turn off buffering on output pipe

print $output "received $_" while <$input>;

close $input;
close $output;

File: golang/streaming_pipe_in_and_out.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

func main() {
	input, _ := os.Open("streaming-in.pipe")
	defer input.Close()

	output, _ := os.OpenFile("streaming-out.pipe", os.O_WRONLY, 0)
	defer output.Close()

	output.Sync() // turn off buffering on the ouput pipe

	scanner := bufio.NewScanner(input)

	for scanner.Scan() {
		line := scanner.Text()
		fmt.Fprintf(output, "received %s\n", line)

	}
}
